### PR TITLE
[Docs] Updates README.md

### PR DESCRIPTION
**Pull request by mnigh**
## 👋 Introduction

This PR updates the root `README.md` file.

## 🕵️ Details

- links `maintenance/README.md` using markdown
- changes a few sentences for consistency
- removes **tc-report** historic details

### CODE REVIEW

**Comment by esizer** [comment]
Why is typescript styled with strikeout? 🤔 Should we remove that as well?

https://github.com/GCTC-NTGC/gc-digital-talent/blob/bb7ce543298858367e1d8e4568faeabd9b23d7f8/README.md?plain=1#L34

**Comment by mnigh** [comment]
> Why is typescript styled with strikeout? 🤔 Should we remove that as well?
> 
> https://github.com/GCTC-NTGC/gc-digital-talent/blob/bb7ce543298858367e1d8e4568faeabd9b23d7f8/README.md?plain=1#L34

I have always wondered that myself. Maybe it was meant to communicate that TypeScript is not covered by CodeQL? Does any of the @GCTC-NTGC/developers know? It was added [here](https://github.com/GCTC-NTGC/gc-digital-talent/commit/a4b0c8355dd207dcdcec5870d79dc7df93f3f565#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R31) as part of https://github.com/GCTC-NTGC/gc-digital-talent/pull/2719.

According to multiple documentation sources, TypeScript **is** covered so maybe things have changed since May 2022:

- https://docs.github.com/en/code-security/code-scanning/automatically-scanning-your-code-for-vulnerabilities-and-errors/customizing-code-scanning#changing-the-languages-that-are-analyzed
- https://codeql.github.com/docs/codeql-overview/supported-languages-and-frameworks/#id23

File: README.md [hunk 1]
```diff
@@ -4,13 +4,13 @@
...
 The GC Digital Talent app is divided into multiple services, each treated as its own sub-project:
 
-- `/api`, the API service
-- `/apps`, (new) location for frontend applications
+- `/api`, API service
+- `/apps`, frontend applications
 - `/packages`, npm packages used within `/apps`
 - `/apps/e2e`, e2e testing with [`Cypress`](/apps/e2e/cypress/README.md)
-- `/tc-report`, containing static content generated with Jekyll, mostly the Talent Cloud report. This used to be maintained in a [separate repo](https://github.com/GCTC-NTGC/tc-report), but now the entire project has been imported into this repo. See [`/tc-report/README.md`](/tc-report/README.md) for instructions on how to run Jekyll and re-generate the site content.
+- `/tc-report`, static content, mostly the Talent Cloud report, generated with [`Jekyll`](/tc-report/README.md)
```
Comment (petertgiles) [inline README.md:11]: It feels a little funny to have a link with the text "Jekyll" go to our tc report readme.  I'm not sure what the link text should be.  "Read more"?

Comment (mnigh) [reply README.md:11]: Ya, I was following the convention that `Cypress` started, but am open to removing both! Thoughts?

Comment (petertgiles) [reply README.md:11]: Yeah, the Cypress page is actually about Cypress though  (despite the H1 title on the page  :laughing: ) so I think that one's OK.  

Maybe the directory name could be the link?  
```
[`/tc-report`](/tc-report/README.md), static content, mostly the Talent Cloud report, generated with Jekyll
```

Comment (mnigh) [reply README.md]: Please see a85d688962eb27268d100b63b6d866a9e99d0954.

**Review by petertgiles — APPROVED** [review]
LGTM.
I vote to un-strikeout (strike-in?) Typescript.

**Review by esizer — APPROVED** [review]
🦃

**Pull request merged**